src/classes/tmControllerTrialFeatureRequest.cls

```diff
@@ -0,0 +1,486 @@
+        } catch (Exception ex) {
+            showForm = false; 
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.FATAL, ex.getMessage()));
+            tmOrgLog.log( BaseOrgLog.level.INFO,'tmControllerTrialFeatureRequest.tmControllerTrialFeatureRequest',ex.getMessage(),tfrId);
+        }
+    }
+
+    /*
+    * Loads countries from Country Revenue Map object
+    */
+    private List<sfbase__CountryRevenueMap__c> countryList {
+        get {
+            if (countryList == null) {
+                countryList = tmDaoProvisioning.getISOCountryList();
+            }
+            return countryList;
+        }
+        private set;
+    }
+
+    public List<SelectOption> countryOptions {
+        get {
+            if (countryOptions == null) {
+                countryOptions = new List<SelectOption>();
+                for (sfbase__CountryRevenueMap__c countryMap : countryList) {
+                    countryOptions.add(new SelectOption(countryMap.sfbase__CountryISOCode__c,countryMap.sfbase__CountryName__c));       
+                }
+            }
+            return countryOptions;
+        }
+        private set;
+    }
+
+    public Boolean isCountryEditable {
+        get {
+            return tfrDomain.isDemoAccount && !tfrDomain.hasValidIsoCountryCode;
+        }
+        private set;
+    }
+
+    public String selectedCountryName {
+        get {
+            for (sfbase__CountryRevenueMap__c countryMap : countryList) {
+                if (tfrDomain.countryCode.equalsIgnoreCase(countryMap.sfbase__CountryISOCode__c)) {
+                    selectedCountryName  = countryMap.sfbase__CountryName__c;
+                    break;
+                }
+            }
+            return selectedCountryName;
+        }
+        private set;        
+    }
+
+    private String objectIdForLog {
+        get {
+
+            return tfrDomain!= null && tfrDomain.tfrDecorator != null && tfrDomain.tfrDecorator.tfr !=null && tfrDomain.tfrDecorator.tfr.Id != null ?  tfrDomain.tfrDecorator.tfr.Id  : tfrAccountId ;
+        }
+        private set;        
+    }
+
+    /*
+    *   Save the TFR and if it is an insert then take the user to the Edit page for that record. If
+    *       The TFR already exists then only refresh page with a return null.
+    */
+    public PageReference saveTfr() {
+        showSaved = true;
+        try {
+            updateUpgradeInfoToTFR();
+            tfrDomain.selectedProducts = orgProducts;
+            if (tfrDomain.tfrDecorator.canEdit) {
+                tfrDomain.save();
+            }
+            
+            if (String.isBlank(tfrId)) {
+                tmTrialFeatureRequest__c t = new tmTrialFeatureRequest__c(Id = tfrDomain.tfrDecorator.tfr.Id);
+                ApexPages.StandardController sc = new ApexPages.StandardController(t);
+                PageReference p = sc.edit();
```

Comment:
This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.